pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -178,20 +193,25 @@ func (exp *flowExporter) initFlowExporter(collector net.Addr) error {
 }
 
 func (exp *flowExporter) sendFlowRecords() error {
-	sendAndUpdateFlowRecord := func(key flowexporter.ConnectionKey, record flowexporter.FlowRecord) error {
-		dataSet := ipfix.NewSet(ipfixentities.Data, exp.templateID, false)
-		if err := exp.sendDataSet(dataSet, record, exp.templateID); err != nil {
+	dataSet := ipfix.NewSet(ipfixentities.Data, exp.templateID, false)
+	addAndUpdateFlowRecord := func(key flowexporter.ConnectionKey, record flowexporter.FlowRecord) error {
+		if err := exp.addRecordToSet(dataSet, record, exp.templateID); err != nil {
```

Comment:
Could we add a check before this if the current length of dataset exceeds the message size limit if the record is added, and then send the dataset here itself before moving on to next record? 
If it is not simple and there is no time for you, please add a TODO. We should handle this in a followup patch.

Reply:
Makes sense. Added.